iot-hub/Samples/device/ConventionBasedOperations/TemperatureController/TemperatureControllerSample.cs

```diff
@@ -91,6 +110,70 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
             }
         }
 
+        private async Task GetWritablePropertiesAndHandleChangesAsync()
+        {
+            ClientProperties properties = await _deviceClient.GetClientPropertiesAsync();
+            ClientPropertyCollection writableProperties = properties.WritablePropertyRequests;
+            long serverWritablePropertiesVersion = writableProperties.Version;
+
+            // Check if the writable property version is outdated on the local side.
+            // For the purpose of this sample, we'll only check the writable property versions between local and server
+            // side without comparing the property values.
+            if (serverWritablePropertiesVersion > s_localWritablePropertiesVersion)
+            {
+                _logger.LogDebug($"The writable property version cached on local is changing from {s_localWritablePropertiesVersion} to {serverWritablePropertiesVersion}.");
+
+                foreach (KeyValuePair<string, object> writableProperty in writableProperties)
+                {
+                    switch (writableProperty.Key)
+                    {
+                        case Thermostat1:
+                        case Thermostat2:
+                            const string targetTemperatureProperty = "targetTemperature";
+                            string componentName = writableProperty.Key;
+
+                            // Comparing here with Line189, the writableProperties variable is obtained by manually calling GetClientPropertiesAsync()
+                            // rather than using property update requests callback, so we are not getting a WritableClientProperty out here.
+                            if (writableProperties.TryGetValue(componentName, targetTemperatureProperty, out double targetTemperatureValue))

```

Comment:
I don't feel confident about explanation in my comments above, so would be open to any suggestions! Same for the thermostat sample as well.

Reply:
How about:
"The `SubscribeToWritablePropertyUpdateRequestsAsync` callback makes the writable property requests available as a `WritableClientProperty`, which provides convenience methods for acknowledging the requests. Since property update requests that were potentially lost during reconnection can be retrieved only as a collection of property values, and not as a collection of `WritableClientProperty`, we will need to create the property response ack by ourselves."

@drwill-ms - thoughts?

Reply:
Modified.